### ISSUE TITLE
goog.module naming module folder name

### ISSUE THREAD

**Issue by appsforartists** [angular/tsickle#252]
If I look at the compiled results of `tsickle`, I'm seeing names like 

```
goog.module('dist.index')
```

in `./dist/index.js`.  I'd expect `tsickle` to parse `package.json` and use `name` and `main` to figure out what the correct module name would be.  For instance, I'd expect this module to be called `material-motion-runtime`.

You can [see my configuration here](https://github.com/material-motion/material-motion-experiments-js/tree/develop/packages/runtime).

**Comment by b-strauss** [comment]
Yes, I just saw that tsickle produces different module names on different systems (windows/mac). Tsickle should use name and main for the module names, with the module's subfolders being appended to that.

**Comment by alexeagle** [comment]
actually, the tsickle design is the other way around - follow closure's scheme of assigning `goog.module` names based on the actual path of the file on disk, but update any `import` statements in the rest of your code to match.
We do this because you might choose to keep ESM module syntax (`import/export`) in the output of tsickle (as Angular does, because we have non-closure users of the library). In that case Closure will always assign the `goog.module` names based on the file path.

So in the original example, anywhere you `import 'material-motion-runtime'` it should be rewritten as `import 'material-motion-runtime/dist/index'`. That doesn't work either right now, #376
